api_app/serializers.py

```diff
@@ -179,10 +194,48 @@ def validate(self, attrs: dict) -> dict:
         file_obj.seek(0)
         file_buffer = file_obj.read()
         attrs["md5"] = calculate_md5(file_buffer)
+        self.filter_analyzers_and_connectors(attrs)
         logger.debug(f"after attrs: {attrs}")
         return attrs
 
 
+class MultipleObservableAnalysisSerializer(rfs.ListSerializer):
+    """
+    ``Job`` model's serializer for Multiple Observable Analysis.
+    Used for ``create()``.
+    """
+
+    def update(self, instance, validated_data):
+        raise NotImplementedError("This serializer does not support update().")
+
+    child = None
+    many = True
+    observables = rfs.ListField(required=True)
+
+    def to_internal_value(self, data):
```

Comment:
also, why did you use `to_internal_value` instead of `validate`? If possible, it would be better to use `validate` cause it works on a higher level so it is less bug prone.

From the docs: ```If you don't need to alter deserialization behavior and instead want to provide object-level validation, it's recommended that you instead override the [.validate()](https://www.django-rest-framework.org/api-guide/serializers/#object-level-validation) method```

Reply:
`ListSerializer.to_internal_value` needs to be overriden as it has assertion(s) that would fail due to our payload not being of `[{...item1}, {...item2}]` style. For instance, this check:
```
        if not isinstance(data, list):
            message = self.error_messages['not_a_list'].format(
                input_type=type(data).__name__
            )
            raise ValidationError({
                api_settings.NON_FIELD_ERRORS_KEY: [message]
            }, code='not_a_list')
```

Reply:
ok that makes sense. But, on another thought, I think that it would be simpler and more linear to change how we pass the parameters instead of changing how the framework accept the data. could it be something like 
```
    "observables": [{"classification": "ip", "name":"8.8.8.8"}, ...],
```
?

Reply:
That would lead to quite a lot of duplication

Reply:
```
{
    "observables": [["ip", "8.8.8.8"], ["domain", "example.com"]],
    "analyzers_requested": [
        "Classic_DNS",
        "Robtex_IP_Query"
    ],
    "connectors_requested": [],
    "tlp": "WHITE",
    "runtime_configuration": {},
    "tags_labels": []
}
``` 
becomes
```
[
  {
      "observable_name": "8.8.8.8",
      "observable_classification" : "ip",
      "analyzers_requested": [
          "Classic_DNS",
          "Robtex_IP_Query"
      ],
      "connectors_requested": [],
      "tlp": "WHITE",
      "runtime_configuration": {},
      "tags_labels": []
  },
{
      "observable_name": "example.com",
      "observable_classification" : "domain",
      "analyzers_requested": [
          "Classic_DNS",
          "Robtex_IP_Query"
      ],
      "connectors_requested": [],
      "tlp": "WHITE",
      "runtime_configuration": {},
      "tags_labels": []
  },
]
```